imap/http_caldav_sched.c

```diff
@@ -2474,14 +2474,14 @@ static void update_attendee_status(icalcomponent *ical, strarray_t *onrecurids,
 
 static icaltimetype get_historical_cutoff()
 {
-    int age = config_getint(IMAPOPT_CALDAV_HISTORICAL_AGE);
+    int age = config_getduration(IMAPOPT_CALDAV_HISTORICAL_AGE, 'd');
     icaltimetype cutoff;
 
     if (age < 0) return icaltime_null_time();
 
     /* Set cutoff to current time -age days */
     cutoff = icaltime_current_time_with_zone(icaltimezone_get_utc_timezone());
-    icaltime_adjust(&cutoff, -age, 0, 0, 0);
+    icaltime_adjust(&cutoff, 0, 0, 0, -age);
```

Comment:
libical is aware of the day of month, so adjusting with seconds here isn't them same as adjusting for days. I don't think it makes much of a difference for the use case, but you might want to check with @murch if we want to keep this option a days-bound integer value.

Reply:
I don't think this is a problem, but we need to verify that icaltime_adjust() can properly handle a large number of seconds (e.g. 365 days)

Reply:
I just ran a quick test and icaltime_adjust() works fine with -365 days worth of seconds.

Reply:
Thanks for checking that, I meant to ask about it but then forgot!

Reply:
I'm really not sure whether to change this one back to simple int days.  All the other DURATIONs are things that, one way or another, were being converted down to simple seconds and then compared to time_t's.  But this bit is actually capable of doing real d/h/m/s calculations, so it seems a bit weird to squish it down to seconds the simple way and fudge it through like that.

Reply:
I tend to agree with your sentiment, but I'm fine with it either way.